File: LeetCode-Python/SummaryRanges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        break_point = 0
        k = 0
        res = []

        if len(nums) == 1:
            res.append(f"{nums[0]}")

        for i in range(1, len(nums)):
            
            if nums[i] - nums[k] > 1:
                if nums[break_point] == nums[k]:
                    res.append(f"{nums[i - 1]}")
                else:
                    res.append(f"{nums[break_point]}->{nums[k]}")
                break_point = i


            if i == len(nums) - 1:
                if nums[break_point] == nums[i]:
                    res.append(f"{nums[i]}")
                else:
                    res.append(f"{nums[break_point]}->{nums[i]}")

            k += 1

        return res

File: LeetCode-Cpp/SummaryRanges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        // Create a list of string to store the output result...
        vector<string> output;
        // Start traversing the array from idx = 0 till idx < sizeofarray in a while loop.
        int idx = 0;
        while(idx < nums.size()) {
            // Initialize beg and last index for identifying the continuous element in the array...
            int beg, last;
            // Mark the number at current index as beginning element of the range...
            beg = nums[idx];
            // Traverse the array beggining from current index & find the last element whose difference from previous element is exactly 1, i.e. nums[idx + 1] == nums[idx] + 1...
            while(idx+1 < nums.size() && nums[idx+1] == nums[idx] + 1)
                idx++;
            // Set this element as last element of the range...
            last = nums[idx];
            // If continuous element isn't present...
            if(beg == last)
                output.push_back(to_string(beg));
            // If present...
            else
                output.push_back(to_string(beg) + "->" + to_string(last));
            idx++;
        }
        return output;      // Return the output result list
    }
};
